packages/@coorpacademy-app-player/src/store/view/state-to-props/player.js

```diff
@@ -48,21 +48,29 @@ const getProgressionStep = state => {
     : null;
 };
 
+const _getRoute = (state, slideContext) => {
+  const route = getRoute(state);
+  if (includes(route, ROUTES)) {
+    return route;
+  }
+  return slideContext ? 'context' : 'answer';
```

Comment:
pk vous avez pas fait: Tant que le determination de la route(onglet) n'est pas fait (cf : https://github.com/CoorpAcademy/components/blob/contextbeforeslide/packages/@coorpacademy-app-player/src/store/actions/ui/progressions.js#L63) je retourne `{}` (état loading)

Ce qui me dérange dans cette approche c'est que la logique de `getRoute` est développée 2 fois et de 2 manières différentes:
`src/store/actions/ui/progressions.js`:
```
if (isNil(get('context.title', slideResult))) {
        return dispatch(selectRoute('answer'));
      }
else return dispatch(selectRoute('context'));
```
ici:
```
if(get('context', slide))
return `context`;
else return `answer`;
```

Reply:
On aurait effectivement peut-être pu faire ça plutôt. Par contre, ça ralentirait (de quelques dixièmes de secondes) l'affichage de la page, en gros le temps que la route answer est (actuellement en prod) affiché avant de switcher vers le contexte. Mais je suis d'accord que ça pourrait être pas mal de plutôt résoudre à un seul endroit.

Mais du coup ce sera à quel endroit qu'il faudrait retourner `{}` pour avoir un état de loading ?

Reply:
@jfmengels je pense que tant que l'onglet à sélectionner n'est pas déterminé(pas de `dispatch(selectRoute('...'))`), on doit être en état `loading`. Du coup si dans `state-to-props` du `player`,  `route` n'est pas déterminé on doit retourner `{}`